### PR TITLE
feat(PE-6523): send state to registry on ACL changes

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: versions.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "txIds": {
+    "1723130661028": "XlMDf1o1zFYWwihBxqmgITRtzvvfzjXTBqG132hpwRQ"
+  }
+}
```
Comment (dtfiedler) [inline versions.json:5]: how is this any more useful than the tag information on the repo?

File: .github/workflows/ant.yaml [hunk 2]
```diff
@@ -41,3 +41,34 @@ jobs:
...
+          echo "::set-output name=TX_ID::$TX_ID"
+
+      - name: Create GitHub tag
+        env:
+          TX_ID: ${{ steps.get_tx_id.outputs.TX_ID }}
+        run: |
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+          git tag -a "$TX_ID" -m "Release version $TX_ID"
+          git push origin "$TX_ID"
```
Comment (dtfiedler) [inline .github/workflows/ant.yaml:74]: we seem to be trying to create a versioning pattern in addition to the registry implementation - can we separate these, they are already separate tickets.

File: src/common/balances.lua [hunk 3]
```diff
@@ -8,15 +8,15 @@ function balances.walletHasSufficientBalance(wallet)
 end
 
 function balances.transfer(to)
-	utils.validateArweaveId(to)
+	assert(type(to) == "string", "'To' must be a string")
```
Comment (dtfiedler) [inline src/common/balances.lua:11]: this seems like another change related to ETH address support that doesn't pertain to registry callback logic we've discussed. please scope this PR accordingly.

File: test/evolve.test.js [hunk 4]
```diff
@@ -39,7 +39,7 @@ describe('aos Evolve', async () => {
       },
       evolveResult.Memory,
     );
-
+    console.dir(result, { depth: null });
```
Comment (dtfiedler) [inline test/evolve.test.js]:
```suggestion
```

File: .github/workflows/ant.yaml (hunk 2 above)
Comment (atticusofsparta) [reply .github/workflows/ant.yaml]: removed

File: versions.json (hunk 1 above)
Comment (atticusofsparta) [reply versions.json]: Only in the respect that the ease of access here is simpler but I take your point.

File: src/common/main.lua [hunk 5]
```diff
@@ -24,6 +24,7 @@ function ant.init()
 	Initialized = Initialized or false
 	-- INSERT placeholder used by build script to inject the appropriate ID
 	SourceCodeTxId = SourceCodeTxId or "__INSERT_SOURCE_CODE_ID__"
+	AntRegistryId = AntRegistryId or ao.env.Process.Tags["ANT-Registry-Id"] or nil
```
Comment (dtfiedler) [inline src/common/main.lua:27]: nit - extra nil not necessary
```suggestion
	AntRegistryId = AntRegistryId or ao.env.Process.Tags["ANT-Registry-Id"]
```

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**